src/main/kotlin/app/CodeLongevity.kt

```diff
@@ -0,0 +1,231 @@
+// Copyright 2017 Sourcerer Inc. All Rights Reserved.
+
+package app
+
+import org.eclipse.jgit.diff.DiffFormatter
+import org.eclipse.jgit.diff.DiffEntry
+import org.eclipse.jgit.diff.RawText
+import org.eclipse.jgit.internal.storage.file.FileRepository
+import org.eclipse.jgit.lib.ObjectId
+import org.eclipse.jgit.lib.ObjectLoader
+import org.eclipse.jgit.lib.Repository
+import org.eclipse.jgit.revwalk.RevCommit
+import org.eclipse.jgit.revwalk.RevWalk
+import org.eclipse.jgit.treewalk.TreeWalk
+import org.eclipse.jgit.util.io.DisabledOutputStream
+
+import java.util.Date
+
+/**
+ * Identifies a code line in a file in a revision.
+ */
+class RevCommitLine(val commit: RevCommit, val file: String, val line: Int)
+
+/**
+ * Represents a code line.
+ *
+ * TODO(Alex): the text arg is solely for testing proposes (remove it)
+ */
+class CodeLine(val from: RevCommitLine, val to: RevCommitLine, val text: String) {
```

Comment:
You have a bunch of related comments: this one, about oldId/newId one, probably the one about the alg is more complicated than it should be and what I'm doing about lines that are not yet deleted and couple more; I'll try to answer them here.

Existing lines are reported too, their age is the head's date minus date when they were inserted. When you got a living line and stored its age somewhere, then you will have to update its age on next iteration. That's why I have oldId and newId: newId in the head from previous iteration matches the oldId in the tail in a next iteration.